sketch-on-map/src/main/java/com/esri/arcgismaps/sample/sketchonmap/MainActivity.kt

```diff
@@ -0,0 +1,299 @@
+     */
+    fun stop(view: View) {
+        // get the geometry from sketch editor
+        val sketchGeometry = geometryEditor?.geometry?.value
+            ?: return showError("Error retrieving geometry")
+
+        if (GeometryBuilder.builder(sketchGeometry)?.isSketchValid == false) {
+            return reportNotValid()
+        }
+
+        // stops the editing session
+        geometryEditor?.stop()
+
+        // clear button selection
+        resetButtons()
+
+        // create a graphic from the sketch editor geometry
+        val graphic = Graphic(sketchGeometry)
+        // assign a symbol based on geometry type
+        when (sketchGeometry) {
+            is Polygon -> graphic.symbol = fillSymbol
+            is Polyline -> graphic.symbol = lineSymbol
+            is Point, is Multipoint -> graphic.symbol = pointSymbol
+            else -> {}
+        }
+
+        // add the graphic to the graphics overlay
+        graphicsOverlay.graphics.add(graphic)
+        setCurrentSelectionText("Added graphic to map")
+    }
+
+    /**
+     * De-selects all buttons.
+     */
+    private fun resetButtons() {
+        activityMainBinding.pointLinePolygonToolbar.apply {
+            pointButton.isSelected = false
+            multipointButton.isSelected = false
+            polylineButton.isSelected = false
+            polygonButton.isSelected = false
+            freehandLineButton.isSelected = false
+            freehandPolygonButton.isSelected = false
+        }
+    }
+
+    /**
+     * Clear the MapView of all the graphics and reset selections
+     */
+    fun clear(view: View) {
+        resetButtons()
+        geometryEditor?.clearGeometry()
+        geometryEditor?.clearSelection()
+        setCurrentSelectionText("Cleared")
+    }
+
+    /**
+     * Called if sketch is invalid. Reports to user why the sketch was invalid.
+     */
+    private fun reportNotValid() {
+        val geometry = geometryEditor?.geometry?.value ?: return showError("Geometry not found")
+        val validIfText: String =
+            when (geometry) {
+                is Point -> getString(R.string.invalid_point_message)
+                is Multipoint -> getString(R.string.invalid_multipoint_message)
+                is Polyline -> getString(R.string.invalid_polyline_message)
+                is Polygon -> getString(R.string.invalid_polygon_message)
+                else -> getString(R.string.none_selected_message)
+            }
+        setCurrentSelectionText(validIfText)
+    }
+
+    /**
+     * Set the current selection text to the [selectedItem]
+     */
+    private fun setCurrentSelectionText(selectedItem: String) {
+        activityMainBinding.currentSelection.text =
+            getString(R.string.current_selection, selectedItem)
+    }
+
+    private fun showError(message: String) {
+        Log.e(TAG, message)
+        Snackbar.make(mapView, message, Snackbar.LENGTH_SHORT).show()
+    }
+}
```

Comment:
The sample design mentions support for editing existing geometry from the graphics overlay. Does that also need to be added here?

Reply:
While editing the (red) geometry, you can interact with the geometry in the map to edit the changes made, move the geometry or add additional vertexes. Or does it also mean that graphics are applied and committed to the map should also have support for editing capabilities? 

Looking into the iOS sample viewer, this does not seem to be a functionality that's implemented.